model/src/pyrenew/latent/infection_seeding_method.py

```diff
@@ -0,0 +1,51 @@
+from abc import ABCMeta, abstractmethod
+
+import jax.numpy as jnp
+from jax.typing import ArrayLike
+
+# -*- coding: utf-8 -*-
+# numpydoc ignore=GL08
+"""
+Classes for the infection seeding methods
+"""
+
+
+class InfectionSeedMethod(metaclass=ABCMeta):
+    def __init__(self, n_timepoints: int):
+        self.validate(n_timepoints)
+        self.n_timepoints = n_timepoints
+
+    @staticmethod
+    def validate(n_timepoints: int) -> None:
+        assert isinstance(n_timepoints, int), "n_timepoints must be an integer"
+        assert n_timepoints > 0, "n_timepoints must be positive"
+
+    @abstractmethod
+    def seed_infections(self, I0: ArrayLike) -> ArrayLike:
+        pass
+
+
+class SeedInfectionsZeroPad(InfectionSeedMethod):
+    def seed_infections(self, I0: ArrayLike):
+        return jnp.pad(I0, (self.n_timepoints - I0.size, 0))
```

Comment:
`I0` should always be of size 1. Where should I enforce that? Does that mean I should have `I0: float` *or `I0: int` in the function definition?

In other functions that have `I0` as an argument, they expect `ArrayLike`, so maybe this is a NumPyro thing? ex: https://github.com/CDCgov/multisignal-epi-inference/blob/main/model/src/pyrenew/latent/infections.py#L79

Reply:
Noting that we have changed design strategy here to require array-valued `I0` of at least the length of the largest backward looking PMF (generation interval, feedback strength, etc.)

Reply:
I have revised names to make things clearer, but the entire point of this `InfectionSeedMethod` class to create `I0`'s that fulfill this requirement. I am calling `I0`-like things that are inputs to `InfectionSeedMethod`s `I_pre_seed` and `I0`-like-things that are inputs to `sample_infections` (which must be longer than the largest backward-looking PMF) `I_seed`.

On the note of naming, sample_infections should probably be renames something like `compute_future_infections`, to avoid suggesting something stochastic is happening.

Reply:
Good point, will make issue.

Reply:
Addressed in #137